rfcs/0005-pablo-distribution.md

```diff
@@ -0,0 +1,667 @@
+From (1)
+
+First position <img src="0005-pablo-distribution-assets/images/stem-ed2c175456fa8dcae30f92f61b3694ff.png" width="151" height="20" alt="stem ed2c175456fa8dcae30f92f61b3694ff" /></span>
+
+Second position <img src="0005-pablo-distribution-assets/images/stem-706fbeec167ddb5aeb84ef0c7bde2f57.png" width="192" height="20" alt="stem 706fbeec167ddb5aeb84ef0c7bde2f57" /></span>
+
+Summing these positions would give the original position(equation 1) as
+the ratio terms cancel out.
+
+As this method uses a reward pooling based approach to calculate the
+rewards for each staker out of it on-demand, rest of the document refers
+to this as the "reward pooling(**RP**) based approach".
+
+#### 5.3.2. Data Structures
+
+Staking rewards pallet already uses the following fNFT data structure,
+
+    /// staking typed fNFT, usually can be mapped to raw fNFT storage type
+    #[derive(Debug, PartialEq, Eq, Copy, Clone, Encode, Decode, TypeInfo)]
+    pub struct StakingNFT<AccountId, AssetId, Balance, Epoch, Rewards> {
+        /// The staked asset.
+        pub asset: AssetId,
+        /// The original stake this NFT was minted for.
+        pub stake: Balance,
+        /// The reward epoch at which this NFT will start yielding rewards.
+        pub reward_epoch_start: Epoch,
+        /// List of reward asset/pending rewards.
+        pub pending_rewards: Rewards,
+        /// The date at which this NFT was minted.
+        pub lock_date: Timestamp,
+        /// The duration for which this NFT stake was locked.
+        pub lock_duration: DurationSeconds,
+        /// The penalty applied if a staker unstake before the end date.
+        pub early_unstake_penalty: Penalty<AccountId>,
+        /// The reward multiplier.
+        pub reward_multiplier: Perbill,
+    }
+
+This data structure can be modified to support the RP approach by
+adding/adjusting/removing some fields.
+
+    /// This can possibly be renamed as `RewardPoolPosition`
+    pub struct StakingNFT<RewardPoolId, AccountId, AssetId, Balance> {
+        /// [Adding] Reward Pool ID from which pool to allocate rewards for this
+        pub reward_pool_id: RewardPoolId,
+
+        /// The staked asset.
+        pub asset: AssetId,
+        /// The original stake this NFT was minted for.
+        /// Used for calculating the pool share.
+        pub stake: Balance,
+
+        /// [Adding] Pool share received for this NFT
+        pub share: Balance,
+
+        /// [Adding] reduced rewards by asset for the fNFT (d_n)
+        pub reductions: BtreeMap<AssetId, Balance>,
+
+        /// [Removing] The reward epoch at which this NFT will start yielding rewards. This is no longer needed.
+        /// pub reward_epoch_start: Epoch,
+
+        /// [Removing] List of reward asset/pending rewards. This is no longer needed as pending rewards are pooled
```

Comment:
already removed. not sure why merge something which is not here

Reply:
You changed the existing code according to new design we agreed, this document is the design that proposes changes to previous version, so we keep it as it is